service/src/main/java/org/databiosphere/workspacedataservice/dao/RecordDao.java

```diff
@@ -144,6 +150,18 @@ public void batchUpsert(UUID instanceId, String recordType, List<Record> records
 		}
 	}
 
+	public void deleteSingleRecord(UUID instanceId, String recordType, String recordId) {
+		try {
+			namedTemplate.getJdbcTemplate().batchUpdate("delete from " + getQualifiedTableName(recordType, instanceId)
+					+ " where " + RECORD_ID + "= " + singleQuote(recordId));
+		} catch (DataIntegrityViolationException e) {
+			if (e.getRootCause()instanceof SQLException sqlEx) {
+				checkForTableRelation(sqlEx);
+				throw e;
+			}
+		}
+	}
+
```

Comment:
```suggestion
	public boolean deleteSingleRecord(UUID instanceId, String recordType, String recordId) {
		try {
			return namedTemplate.update("delete from " + getQualifiedTableName(recordType, instanceId)
					+ " where " + RECORD_ID + " = :recordId", new MapSqlParameterSource("recordId", recordId)) == 1;
		} catch (DataIntegrityViolationException e) {
			if (e.getRootCause()instanceof SQLException sqlEx) {
				checkForTableRelation(sqlEx);
			}
			throw e;
		}
	}

```
This approach switches to named SQL parameters (Sonar will be happier), eliminates the need for the singleQuote function and won't silently capture a DataIntegrityViolationException and it will return true when the record was found and deleted, false if no records were affected (aka record was not found).